app/lib/file_parse_service.rb

```diff
@@ -78,6 +78,16 @@ def self.run_parse_job(study_file, study, user, reparse: false, persist_on_fail:
       when 'Gene List'
         ParseUtils.delay.initialize_precomputed_scores(study, study_file, user)
       when 'Metadata'
+        # log convention compliance -- see SCP-2890
+        if !study_file.use_metadata_convention
+          if User.feature_flag_for_instance(user, 'convention_required')
+            raise 'Metadata files must comply with the SCP metadata convention'
```

Comment:
Are we sure we want to raise an error here?  I'm not sure that would be handled, and would likely show the "Something went wrong" page for users.

Reply:
Good point -- I could see this going either way.  The reason I put this here is that it seemed like the best way to make sure you couldn't get around the requirement by using, e.g. manage-study, or javascript form shenanigans.  You're right that I don't expect the error to be pretty, but I think that's ok since you shouldn't get this error unless you're doing something weird